### DISCUSSION TITLE
Multiple errors from trivy scan

### DISCUSSION THREAD

**Discussion by surajkoditala** [aquasecurity/trivy#7751] [Bugs]
### Description

Azure devops extension - https://learn.microsoft.com/en-us/azure/defender-for-cloud/azure-devops-extension

We are using the below task for trivy scan in the azure pipelines using microsoft security devops. Facing a couple of scenarios and having the problems,

1. The task is successful but don't find the results under the scan tab
2. sometimes getting this error - 
    2024-10-16T15:15:21Z	FATAL	Fatal error	init error: DB error: failed to download vulnerability DB: database download error: oci download error: failed to fetch the layer: GET https://ghcr.io/v2/aquasecurity/trivy-db/blobs/sha256:2ea6fd2ba7b1b63c15bc8b71e78dcb6feb7719f911fe96ae0c6610a10ed11bdc: TOOMANYREQUESTS: retry-after: 380.903µs, allowed: 44000/minute

##[error]Error running trivy job: 1 of 1
##[error]GuardianErrorExitCodeException: trivy completed with an Error exit code: 1. The tool did not complete successfully due to bad parameters or a scan error. Contact TrivySecSupport for assistance.

    - task: MicrosoftSecurityDevOps@1
      displayName: 'Microsoft Security DevOps Container Image Scanning'
      inputs:    
          tools: 'trivy' #string. Optional. A comma-separated list of analyzer tools to run. Values: 'bandit', 'binskim', 'checkov', 'eslint', 'templateanalyzer', 'terrascan', 'trivy'. Example 'templateanalyzer, trivy'
          artifactName: 'CodeAnalysisLogs'

### Desired Behavior

If the task is successful, should be able to see the vulnerabilities under scan tab in each pipeline

### Actual Behavior

No vulnerabilites are being reported or results are shown

### Reproduction Steps

```bash
1.Trigger the azure pipeline with the below task -

    - task: MicrosoftSecurityDevOps@1
      displayName: 'Microsoft Security DevOps Container Image Scanning'
      inputs:    
          tools: 'trivy' #string. Optional. A comma-separated list of analyzer tools to run. Values: 'bandit', 'binskim', 'checkov', 'eslint', 'templateanalyzer', 'terrascan', 'trivy'. Example 'templateanalyzer, trivy'
          artifactName: 'CodeAnalysisLogs'
```

### Target

Container Image

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
N/A
```

### Operating System

Ubuntu

### Version

```bash
MicrosoftSecurityDevOps (1.12.2)
```

### Checklist

- [ ] Run `trivy clean --all`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
Here is the summary.
https://github.com/aquasecurity/trivy/discussions/7699
